### PR TITLE
merging updated code

### PR DESCRIPTION
**Pull request by nikzei**

### CODE REVIEW

File: heroku.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+rm -f .gitignore
+
+mkdir -p .ssh
+echo "$SSH_PRIVATE_KEY" > .ssh/id_rsa
+chmod 600 .ssh/id_rsa
+echo "$SSH_PUBLIC_KEY" > .ssh/id_rsa.pub
+echo -e "Host github.com\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config
+
+git init
+git config --global user.email 'nikzei@gmail.com'
```
Comment (rcackerman) [inline heroku.sh]: Can you re-submit a PR with a sample `heroku.sh` file, instead of a working one?

File: scrape.py [hunk 2]
```diff
@@ -290,12 +310,13 @@ def print_data(parolees):
...
 
 
+# pylint: disable=too-many-locals
 def scrape(old_data_path, no_download):
     """
     Main function -- read in existing data, scrape new data, merge the two
     sets, and save to the output location.
     """
-    scraper = scrapelib.Scraper(requests_per_minute=180,
+    scraper = scrapelib.Scraper(requests_per_minute=1000,
```
Comment (rcackerman) [inline scrape.py]: Why bump up the requests? Our initial thought was to keep it low so it'd be background noise.

File: scrape.py [hunk 3]
```diff
@@ -2,21 +2,25 @@
...
 import sys
-import argparse
-from bs4 import BeautifulSoup
+
+from datetime import datetime
 from string import ascii_uppercase
 from time import localtime, mktime
-from datetime import datetime
+from threading import Thread
+from Queue import Queue
```
Comment (rcackerman) [inline scrape.py]: Yay concurrency!

**Comment by rcackerman** [comment]
Apologies - I **just** saw this today. I left some comments - otherwise 👍 👍 .

**Comment by rcackerman** [comment]
I'm getting the error
```
Traceback (most recent call last):
  File "scrape.py", line 382, in <module>
    scrape(ARGS.input, ARGS.no_download)
  File "scrape.py", line 370, in scrape
    existing_parolees[key].update(parolee)
KeyError: (u'02A6328', '2016-04-26')
```
both after I tried merging and after I cloned your fork and tried from master.

File: scrape.py [hunk 4]
```diff
@@ -47,10 +51,12 @@ def baseurls():
...
     """
     today = localtime()
-    for monthdiff in xrange(-25, 7):
+    #for monthdiff in xrange(-25, 7):
+    for monthdiff in xrange(-2, -1):
         year, month = localtime(mktime(
             [today.tm_year, today.tm_mon + monthdiff, 1, 0, 0, 0, 0, 0, 0]))[:2]
-        for letter in ascii_uppercase:
+        #for letter in ascii_uppercase:
+        for letter in 'A':
```
Comment (rcackerman) [inline scrape.py]: Was this for debugging?

**Comment by rcackerman** [comment]
@talos I fixed the above error (I think you had commented out line 59 for debugging), but I'm still getting errors when the scraper finds a new parolee that's not in the existing parolee list. Looking at `scrape()`, I'm not seeing where you're appending parolees that are completely new... am I missing something?

**Comment by thejunglejane** [comment]
8d17aef5c should resolve the new parolee errors.

**Pull request merged**